container-search/src/main/java/com/yahoo/search/ranking/RankProfilesEvaluator.java

```diff
@@ -50,4 +58,49 @@ public Model modelForRankProfile(String rankProfile) {
     public FunctionEvaluator evaluatorForFunction(String rankProfile, String functionName) {
         return modelForRankProfile(rankProfile).evaluatorOf(functionName);
     }
+
+    static record GlobalPhaseData(Supplier<FunctionEvaluator> functionEvaluatorSource,
+                                  int rerankCount,
+                                  List<String> needInputs) {}
+
+    private Map<String, GlobalPhaseData> profilesWithGlobalPhase = new HashMap<>();
+
+    GlobalPhaseData getGlobalPhaseData(String rankProfile) {
```

Comment:
Consider using `Optional<GlobalPhaseData>` as return type to signify that "empty" is a valid return value